web/satellite/src/components/dashboard/account/AccountArea.vue

```diff
@@ -157,21 +193,268 @@ import HeaderedInput from '@/components/common/HeaderedInput.vue';
+
+                oldPasswordError:"",
+                newPasswordError:"",
+                confirmationPasswordError:"",
+                isPasswordEditing: false
+            }
+        },
+        methods: {
+            setFirstName: function (value: string) {
+                this.$data.firstName = value;
+                this.$data.firstNameError = "";
+                this.$data.isAccountSettingsEditing = true;
+            },
+            setLastName: function (value: string) {
+                this.$data.lastName = value;
+                this.$data.lastNameError = "";
+                this.$data.isAccountSettingsEditing = true;
+            },
+            setEmail: function (value: string) {
+                this.$data.email = value;
+                this.$data.emailError = "";
+                this.$data.isAccountSettingsEditing = true;
+            },
+            setTermsAccepted: function (value: boolean) {
+                this.$data.isTermsAccepted = value;
+                this.$data.isTermsAcceptedError = false;
+            },
+            onCancelAccountSettingsButtonClick: function () {
+                this.$data.firstName = this.$data.cachedFirstName;
+                this.$data.firstNameError = "";
+                this.$data.lastName = this.$store.getters.user.lastName;
+                this.$data.lastNameError = "";
+                this.$data.email = this.$data.cachedEmail;
+                this.$data.emailError = "";
+
+                this.$refs.firstNameInput.setValue(this.$data.cachedFirstName);
+                this.$refs.lastNameInput.setValue(this.$data.cachedLastName);
+                this.$refs.emailInput.setValue(this.$data.cachedEmail);
+
+                this.$data.isAccountSettingsEditing = false;
+            },
+            onSaveAccountSettingsButtonClick: async function () {
+                let hasError = false;
 
+                if (!this.$data.firstName) {
+                    this.$data.firstNameError = "First name expected";
+                    hasError = true;
+                }
+
+                if (!this.$data.lastName) {
+                    this.$data.lastNameError = "Last name expected";
+                    hasError = true;
+                }
+
+                if (!this.$data.email) {
+                    this.$data.emailError = "Email expected";
+                    hasError = true;
+                }
+
+                if (!this.$data.isTermsAccepted) {
+                    this.$data.isTermsAcceptedError = true;
+                    hasError = true;
+                }
+
+                if (hasError) {
+                    return;
+                }
+
+                let user = {
+                    id: this.$store.getters.user.id,
+                    email: this.$data.email,
+                    firstName: this.$data.firstName,
+                    lastName: this.$data.lastName,
+                };
+                await this.$store.dispatch("updateBasicUserInfo", user);
```

Comment:
you should receive some success indicator after dispatching 

let isSuccess = await this.$store.dispatch("updateBasicUserInfo", user);
if (!isSuccess) {
    show error popup
}